apps/veil/src/entities/liquidity/ui/grouped-liquidity-row.tsx

```diff
@@ -0,0 +1,96 @@
+import { useEffect, useState } from 'react';
+import { useParams } from 'next/navigation';
+import { ChevronDown, ChevronUp } from 'lucide-react';
+import { TableCell } from '@penumbra-zone/ui/TableCell';
+import { Button } from '@penumbra-zone/ui/Button';
+import { Text } from '@penumbra-zone/ui/Text';
+import { useRegistry } from '@/shared/api/registry';
+import { DisplayLP } from '../model/get-display-lps';
+import { LiquidityRow } from './liquidity-row';
+import Image from 'next/image';
+
+interface GroupedLiquidityRowProps {
+  pair: string;
+  lps: DisplayLP[];
+}
+
+export const GroupedLiquidityRow = ({ pair, lps }: GroupedLiquidityRowProps) => {
+  const [base, quote] = pair.split('/');
+  const params = useParams<{
+    baseSymbol: string;
+    quoteSymbol: string;
+  }>();
+  const baseSymbol = params?.baseSymbol ?? '';
+  const quoteSymbol = params?.quoteSymbol ?? '';
+  const [isExpanded, setIsExpanded] = useState(false);
+  const toggleExpanded = () => setIsExpanded(!isExpanded);
+
+  useEffect(() => {
+    if (baseSymbol === base && quoteSymbol === quote) {
+      setIsExpanded(true);
+    }
+  }, [baseSymbol, quoteSymbol, base, quote]);
+
+  const { data: registry } = useRegistry();
+  const baseAsset = registry
+    .getAllAssets()
+    .find((asset: { symbol: string }) => asset.symbol === base);
+  const quoteAsset = registry
+    .getAllAssets()
+    .find((asset: { symbol: string }) => asset.symbol === quote);
```

Comment:
discussion: while it's fine for this initial UI phase of the feature, it's best to prefer `useGetMetadata` hook for searching for assets efficiently. But maybe this specific component simply needs to accept the asset per this page, so the assets should be passed as `Metadata`  props?